bindings/java/src/main/native/Makefile.am

```diff
@@ -0,0 +1,55 @@
+topdir=$(abs_top_builddir)
+javadir=$(topdir)/bindings/java
+if JAVA_GUESS
+MVN=@-cd $(javadir) && mvn
+else
+MVN=@cd $(javadir) && mvn
+endif
+
+lib_LTLIBRARIES = libjucx.la
+
+libjucx_la_CPPFLAGS = -I$(JDK)/include -I$(JDK)/include/linux \
+                      -I$(topdir)/src
+
+libjucx_la_SOURCES = bridge.cc request_util.cc
+
+libjucx_la_CXXFLAGS = -fPIC -DPIC -Werror -std=c++0x
+
+libjucx_la_LIBADD = $(topdir)/src/ucs/libucs.la \
+                    $(topdir)/src/uct/libuct.la \
+                    $(topdir)/src/ucm/libucm.la \
+                    $(topdir)/src/ucp/libucp.la
+
+libjucx_la_DEPENDENCIES = Makefile.am Makefile.in Makefile
+
+
+#Compile Java source code and pack to jar
+if JAVA_GUESS
+all:
+	$(MVN) package ; \
+	if [ $$? -ne 0 ] ; \
+        then echo "WARNING: Java support disabled. Execute ./configure --with-java for compilation error." ; \
+        fi
+else
+all:
+	$(MVN) package ; \
+        if [ $$? -ne 0 ] ; \
+        then echo "ERROR: Java support disabled. Exiting..." ; exit 1 ; \
```

Comment:
this should fail during ./configure, not build. in case of guess, if hava is not found, this Makefile should not be used at all.

Reply:
@yosefe 
I checked (in case of guess and didn't find java and mvn) and it didn't fail (during ./configure).
In that case the Makefile is created during the configure, but it's not used during 'make'. What do you mean by "Makefile should not be used"? Do you want it will not be created at all (at the configure phase when HAVE_JAVA == false), or is the current implementation OK (given that it doesn't fail during configure)?

Reply:
Yes, if MVN is not found, it should be as HAVE_JAVA == false.
There should not be any checks in the Makefile itself, all environment checks should be done during ./configure.

Reply:
@miked-mellanox @yosefe @gmegan 
I have 2 suggestions on how to check for Maven availability in the configure (after checking that Java and Maven are installed):
1. Quickest - 'ping <mvn-repos-url>' and check the return code (very quick response).
2. Executing 'mvn dependency:go-offline' so that Maven will try to pull a few repositories needed for the build (could take a very long time).

What's your opinion?

Reply:
can't you just try to compile some dummy test.java with maven and see if it works?

Reply:
it takes extra minutes and acceptable (IMO)

Reply:
why not just try to compile something with maven and see if it works?

Reply:
That's what I did, and it works - meaning it's a good indication weather I can build the Java using Maven or not. So to be clear - in the config I should compile the dummy project, if it succeeds (may take time due to maven downloads), then build Java UCX, correct?

Reply:
yes